### PR TITLE
Add validation method to auth attribute handler

### PR DESCRIPTION
**Pull request by janakamarasena**
This PR introduces a validation method to auth attribute handler. 

Related to: https://github.com/wso2/product-is/issues/15449

### CODE REVIEW

File: components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java [hunk 1]
```diff
@@ -89,6 +92,39 @@ public List<AuthAttributeHolder> getAvailableAuthAttributeHolders(String appId,
         return selectedAuthAttributeHolders;
     }
 
+    /**
+     * This method is used to engage the validations related to the
+     * selected auth attribute handler.
+     *
+     * @param authAttributeHandlerName Name of the auth attribute handler.
+     * @param attributeMap Set of attributes to be validated.
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java:100]: check formatting

File: components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java [hunk 2]
```diff
@@ -89,6 +92,39 @@ public List<AuthAttributeHolder> getAvailableAuthAttributeHolders(String appId,
...
+     * @throws AuthAttributeHandlerException authAttributeHandlerException.
+     */
+    public ValidationResult validateAuthAttributes(String authAttributeHandlerName, Map<String, String> attributeMap)
+            throws AuthAttributeHandlerException {
+
+        List<AuthAttributeHandler> authAttributeHandlers =
+                AuthAttributeHandlerServiceDataHolder.getInstance().getAuthAttributeHandlers();
+
+        AuthAttributeHandler authAttributeHandler = null;
+        for (AuthAttributeHandler handler : authAttributeHandlers) {
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java:113]: can `authAttributeHandlers` be null in any case?

Comment (janakamarasena) [reply components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java:113]: Nope the list is initialized as an empty list in the very beginning so it cannot be null https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/internal/AuthAttributeHandlerServiceDataHolder.java#L34

File: components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java (hunk 1 above)
Comment (janakamarasena) [reply components/org.wso2.carbon.identity.auth.attribute.handler/src/main/java/org/wso2/carbon/identity/auth/attribute/handler/AuthAttributeHandlerManager.java]: fixed

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**